cli/azd/pkg/auth/manager.go

```diff
@@ -590,6 +596,62 @@ func (m *Manager) newCredentialFromFederatedTokenProvider(
 		}
 
 		return cred, nil
+	case oidcFederatedTokenProvider:
+		idToken, idTokenHas := os.LookupEnv("AZURE_OIDC_TOKEN")
+		if idTokenHas {
+			cred, err := azidentity.NewClientAssertionCredential(
+				tenantID,
+				clientID,
+				func(ctx context.Context) (string, error) {
+					return idToken, nil
+				},
+				&azidentity.ClientAssertionCredentialOptions{
+					ClientOptions: clientOptions,
+				})
+			if err != nil {
+				return nil, fmt.Errorf("creating credential: %w", err)
+			}
+
+			return cred, nil
+		}
+
+		token, tokenHas := os.LookupEnv("AZURE_OIDC_REQUEST_TOKEN")
+		tokenUrl, tokenUrlHas := os.LookupEnv("AZURE_OIDC_REQUEST_URL")
+
+		if !idTokenHas && !tokenHas && !tokenUrlHas {
+			//nolint:lll
+			return nil, errors.New(`missing required values. Please set either:
+- AZURE_OIDC_TOKEN (most common in typical setups; provide your ID token directly), OR
+- Both AZURE_OIDC_REQUEST_TOKEN and AZURE_OIDC_REQUEST_URL (used when your CI provider requires an additional token exchange; less common)`)
```

Comment:
should we add this to the `azd login --help` ?  or should we create a an entry in docs about this?

Reply:
We'll need some docs update -- not planning to update the `login --help` since this is an advanced scenario in CI.

I'll create one.

Reply:
Created #5417 cc: @kristenwomack